Merge objective options into scenario options

AaronVanGeffen opened the pull request:
This PR merges the objective options window into the scenario options window.

To do:
- [x] Initial pass: simply merge windows
- [x] Split scenario name/group/details into its own page (hide in-game?)
- [x] Move 'difficult guest generation' setting into main guest options page
- [x] Move 'hard park rating' setting into finances page
- [x] Move 'pay per ride/entrance' setting into finances page
- [x] Change 'park' tab into terraform tab
- [x] Adjust window captions/titles
- [x] Add group boxes to financial options tab
- [x] Add label next to 'pay per ride/entrance' dropdown ('earning mode'? 'business model'?)
- [x] Fix swapped title/tooltip for 'land restriction' tab
- [x] Tweak window/tab dimensions?
- [x] Adjust editor: automatically switch to objective tab

Screenshots:

![scenario-options-1](https://github.com/user-attachments/assets/d67d62ed-55cd-4c99-80d3-61d33d134283)
![scenario-options-2](https://github.com/user-attachments/assets/fd84fd58-6857-4051-ad13-7f4e9dced740)
![scenario-options-3](https://github.com/user-attachments/assets/e2d3e7ab-a9d6-4700-9fd1-3556a71d4706)
![scenario-options-4](https://github.com/user-attachments/assets/7134c3c4-a766-43ce-9f06-4c3a3256d933)
![scenario-options-5](https://github.com/user-attachments/assets/7acfc1db-5105-4ac9-8880-343fdfb87e87)
![scenario-options-6](https://github.com/user-attachments/assets/d3acc22e-6366-4051-987a-fb739e09831c)

Closes #23930

Gymnasiast reviewed (changes requested):
Window does not open in the centre of the screen:

1. ![afbeelding](https://github.com/user-attachments/assets/171c2959-6288-4b10-a4b2-fd604d69890d)
2. Even though "No money" is selected, I am still presented with money-related options: 
![afbeelding](https://github.com/user-attachments/assets/a3f86602-7255-4135-9f38-04ed4d00c119)
3. Conversely, disabling money also hides options unrelated to money, like a forbidden tree removal: 
![afbeelding](https://github.com/user-attachments/assets/02ebd106-7e28-459e-bb6c-f574220e1027)
4. The Scenario Editor presents you with the same window twice.
5. The Scenario Editor does not block saving the scenario if you didn’t fill in a name. Since this option is now no longer automatically visible, I think it should a) warn you about it, and b) change the window to the appropriate tab if it does.

AaronVanGeffen commented:
Thanks for testing. I'll address the issues soon.

> Window does not open in the centre of the screen: 

I'm surprised! I thought the same window flags would apply to both objective and scenario options.

> Even though "No money" is selected, I am still presented with money-related options:

Huh. I recognise this, and thought I fixed this, but clearly not.

> Conversely, disabling money also hides options unrelated to money, like a forbidden tree removal:

Argh, this will be due to them sharing widget indices. 

> The Scenario Editor presents you with the same window twice.

> The Scenario Editor does not block saving the scenario if you didn’t fill in a name. Since this option is now no longer automatically visible, I think it should a) warn you about it, and b) change the window to the appropriate tab if it does.

Yes, this is what I meant by my final to-do item. Rather than merging the editor steps, I'd like it to switch to the right (next/previous) tab in the window at the very least.

AaronVanGeffen commented:
@Gymnasiast I believe I've addressed the issues you pointed out. You can probably deduce what's changed from the commit messages, but for completeness:

- The scenario window is now centred on open
- The issue with 'no money' widgets has been resolved ("Account for finance group boxes having been swapped")
- Deliberate change: the editor now allows you to going back when working with a converted save (as far back as the scenario options, not the invention list)
- The scenario options window now has three stages, all of which are individual editor steps: general options (1), objective selection (2), scenario details (3)

AaronVanGeffen commented:
@Gymnasiast I believe I've addressed the issues you pointed out. You can probably deduce what's changed from the commit messages, but for completeness:

- The scenario window is now centred on open
- The issue with 'no money' widgets has been resolved ("Account for finance group boxes having been swapped")
- The scenario options window now has three stages, all of which are individual editor steps: general options (1), objective selection (2), scenario details (3)
- Deliberate change: the editor now allows you to going back when working with a converted save (as far back as the scenario options, not the invention list)

LordMarcel commented:
Ride preservation tab is in two places: Options selection and objective selection.

![image](https://github.com/user-attachments/assets/b1ed5cd8-29b5-4b5c-a9a4-37dc074fe51a)

![image](https://github.com/user-attachments/assets/d56534f0-259b-49f9-bd24-c0bc62d2e577)

AaronVanGeffen commented:
@LordMarcel The issue with the rides preservation tab should now be resolved. Could you re-check? :)

LordMarcel commented:
It's all good now.

Gymnasiast reviewed (changes requested):
Got around to testing. I think the way you divided it up in three stages makes a lot of sense. I found a few bugs, though.

Overlapping text when RCT1 interest calculation is enabled (should hide the loan percentage label and spinner, but does not appear to hide label)
![afbeelding](https://github.com/user-attachments/assets/2899fedf-4475-4331-b7e9-a7165fb1b91d)

Permanently pressed down `+` button:
![afbeelding](https://github.com/user-attachments/assets/627021b6-5cda-4120-83d9-c502e2a2bf4f)

Disabling money does not hide this label:
![afbeelding](https://github.com/user-attachments/assets/6fbcfcac-aa2e-482e-bf05-83b5bf4060f8)

When clicking `Save scenario` in the last step, the game behaves a bit weird.

For starters, the Back/Forward buttons do not update. If you click ‘Back’, you’re go back to Objective Selection (the step before Scenario Details) and not back to Scenario Details, which was the actual previous step. It will also keep the Load/save window open.

Expectation: clicking ‘Next’ during ‘Scenario details’ stage behaves like an actual step, with the Forwards button being disabled (as it’s the last step), and both the Back button and closing the load/save window taking you back to ‘Scenario Details’.

AaronVanGeffen commented:
Thanks for testing!

> Overlapping text when RCT1 interest calculation is enabled (should hide the loan percentage label and spinner, but does not appear to hide label) 

Thanks. This must've slipped in during the recent rebase. Should be a quick fix.

> Permanently pressed down `+` button:

Interesting! I wonder how that happened. Will fix.

> Disabling money does not hide this label:

Likely an off by one as a result of the new label widgets. Will fix.

> When clicking `Save scenario` in the last step, the game behaves a bit weird.
> 
> For starters, the Back/Forward buttons do not update. If you click ‘Back’, you’re go back to Objective Selection (the step before Scenario Details) and not back to Scenario Details, which was the actual previous step. It will also keep the Load/save window open.

This is true on develop as well. The editor step doesn't seem to update correctly when the modal window is open. This becomes apparent when the 'step label' at the bottom-centre isn't updated to 'Save scenario' either, nor is the 'next' button hidden when you're at this final step.

I've not been able to get to the bottom of what causes this, but I think you're right that this is as good an opportunity as any to fix this. I'll try to dive into this again.

Gymnasiast reviewed (changes requested):
Label is cut off way too early (too small?)

![afbeelding](https://github.com/user-attachments/assets/01689f1d-9a53-47ab-b782-1cebfe473cb6)

When continuing to "save scenario" (in the scenario editor), the bottom toolbar does not invalidate properly (still showing the "objective" step as the previous, which should be "scenario details"):

![afbeelding](https://github.com/user-attachments/assets/a5aa5950-d81d-499d-9d61-71685baa213a)

And when forcing invalidation, the "previous button" just disappears. Rather, I would expect a back button for the scenario details here.

AaronVanGeffen commented:
Thanks for testing.

> Label is cut off way too early (too small?)
> 
> ![afbeelding](https://private-user-images.githubusercontent.com/1478678/434521977-01689f1d-9a53-47ab-b782-1cebfe473cb6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ4MzY5OTMsIm5iZiI6MTc0NDgzNjY5MywicGF0aCI6Ii8xNDc4Njc4LzQzNDUyMTk3Ny0wMTY4OWYxZC05YTUzLTQ3YWItYjc4Mi0xY2ViZmU0NzNjYjYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MTZUMjA1MTMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmI1YTEzOWM3Nzg5MzhhZDhlNTRmNDk1YWViMjdhZGRhZWVmNTkxZTk0NWM3NjA3MGViNWNlZjRhZWM5NjZhNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.l7v6ptIOv0rotBPTpXiOqbUk9qtIxC2Qole-8R6_CJE)

I clearly should have tested in Dutch 😉 Thanks, I've now widened the labels widgets considerably.

> When continuing to "save scenario" (in the scenario editor), the bottom toolbar does not invalidate properly (still showing the "objective" step as the previous, which should be "scenario details"):

I've added an explicit invalidation call. I guess I was using OpenGL when I last tested this, so I hadn't noticed.

> And when forcing invalidation, the "previous button" just disappears. Rather, I would expect a back button for the scenario details here.

Right, I see what you mean. The thing is, the load/save window is a modal window, so the buttons aren't actually triggered. Hiding them is preferable over having non-functional buttons, I think. While perhaps less intuitive, closing the load/save window does automatically return to the previous step.

(They back/forward buttons are both visible on master/develop, but in the non-functional state. They also point to the wrong steps.)
